internal/pkg/service/common/etcdop/watch.go

```diff
@@ -6,31 +6,70 @@ import (
 	"go.etcd.io/etcd/api/v3/etcdserverpb"
 	"go.etcd.io/etcd/api/v3/mvccpb"
 	etcd "go.etcd.io/etcd/client/v3"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+type EventType int32
+
+const (
+	CreateEvent EventType = iota
+	UpdateEvent
+	DeleteEvent
 )
 
 type EventT[T any] struct {
-	*etcd.Event
+	op.KeyValueT[T]
+	Type   EventType
```

Comment:
Funguje to tak, ze najprv pomocou iteratora nacitame vsetko, a generujeme eventy typu `Create`.
A potom spustime standardny watcher, ... tam sa da tiez odlisit `create/update`.

Pouzil som tu `KeyValueT` aby sa to jednoducho napojilo na `iterator` + obsahuje to (value + raw KV) co tam vlastne chceme mat.